lib/bibliothecary/analyser.rb

```diff
@@ -142,15 +142,15 @@ def analyse_contents(filename, contents)
 
       def analyse_contents_from_info(info)
         dependencies = parse_file(info.relative_path, info.contents)
-        if dependencies && dependencies.any?
-          {
-            platform: platform_name,
-            path: info.relative_path,
-            dependencies: dependencies,
-            kind: determine_kind_from_info(info),
-            success: true
-          }
-        end
+        dependencies = [] if dependencies.nil?
```

Comment:
Implication here is that if parsers actually mean "error" they now need to raise Bibliothecary::FileParsingError, and if they actually mean "didn't know how to handle this file after all" they need to screen the file out in match_info. I do think a few of the `return []` cases in the parsers should probably return errors, but returning no deps found isn't a catastrophe, and don't want to go research the schemas of every package manager right now. I don't think think we have any "don't know how to handle this file after all" cases really, but I guess if we do we'll find out.

Reply:
Maybe a comment to that effect here... also are there other non-truthy things that could be returned or could you just do `dependencies || []` below?